jsx/App/Stories/Story/lib/txt_sync.js

```diff
@@ -135,3 +149,45 @@ export function setupTextSync() {
     });
   
 }
+
+
+export function setupYoutube() {
+
+    const youtubeMedia = document.querySelectorAll("[is-youtube='true']")[0];
+    if (!youtubeMedia) {
+        return; 
+    }
+
+    const youtubeID = youtubeMedia.getAttribute('youtube-id');
+
+    // Initialize YouTube player: 
+    window.YT.ready(function() {
+        player = new window.YT.Player("video", {
+            height: "270",
+            width: "480",
+            videoId: youtubeID,
+            events: {
+                'onReady': onPlayerReady,
+                'onStateChange': onPlayerStateChange
+            }
+        });
+    });
+
+    function onPlayerReady(event) {
+    
+    }
+
+    // This function is called when the player's state changes, ie.
+    // when the Youtube player is paused, resumed, finished, etc. 
+    function onPlayerStateChange(event) {
+
+    } 
+
+    // Call the sync function on the Youtube video every second.
+    // This ensures that the corresponding text is highlighted when
+    // the text's timestamp matches the video's timestamp. 
+    setInterval(function(){
```

Comment:
Do we get noticeably better syncing when we call it every 0.1 second? If so, let's do that. 0.1 second is a good interval for UI updates because it appears instantaneous, while usually leaving plenty of time in between updates so the page doesn't become sluggish.

Reply:
(If the sync function is veeeerrrry slow, we would actually get worse performance calling it every 0.1 second. I think this is unlikely, even though sync runs in O(n^2) time where n is the number of sentences.)

Reply:
I will call it every 0.1 seconds now, though I actually don't see too noticeable of a difference between 1 second vs. 0.1 second haha... I think it's because the text's timestamps start at the beginning of the second, so calling this sync function every 0.1 only slightly makes the text highlighting appear faster.